src/control/system/database_pools.go

```diff
@@ -68,6 +83,76 @@ type (
 	}
 )
 
+// NewPoolService returns a properly-initialized *PoolService.
+func NewPoolService(uuid uuid.UUID, rankScm, rankNvme uint64, ranks []Rank) *PoolService {
+	rs := RankSetFromRanks(ranks)
+	return &PoolService{
+		PoolUUID: uuid,
+		State:    PoolServiceStateCreating,
+		Storage: &PoolServiceStorage{
+			ScmPerRank:      rankScm,
+			NVMePerRank:     rankNvme,
+			CreationRankStr: rs.RangedString(),
+			CurrentRankStr:  rs.RangedString(),
+		},
+	}
+}
+
+// CreationRanks returns the set of target ranks associated
+// with the pool's creation.
+func (pss *PoolServiceStorage) CreationRanks() []Rank {
+	pss.Lock()
+	defer pss.Unlock()
+
+	if pss.creationRanks == nil {
+		var err error
+		pss.creationRanks, err = CreateRankSet(pss.CreationRankStr)
+		if err != nil {
+			return nil
+		}
+	}
+	return pss.creationRanks.Ranks()
+}
+
+// CurrentRanks returns the set of target ranks associated
+// with the pool's current.
+func (pss *PoolServiceStorage) CurrentRanks() []Rank {
+	pss.Lock()
+	defer pss.Unlock()
+
+	if pss.currentRanks == nil {
+		var err error
+		pss.currentRanks, err = CreateRankSet(pss.CurrentRankStr)
+		if err != nil {
+			return nil
+		}
+	}
+	return pss.currentRanks.Ranks()
+}
+
+// TotalSCM returns the total amount of SCM storage allocated to
+// the pool, calculated from the current set of ranks multiplied
+// by the per-rank SCM allocation made at creation time.
+func (pss *PoolServiceStorage) TotalSCM() uint64 {
+	return uint64(len(pss.CurrentRanks())) * pss.ScmPerRank
+}
+
+// TotalNVMe returns the total amount of NVMe storage allocated to
+// the pool, calculated from the current set of ranks multiplied
+// by the per-rank NVMe allocation made at creation time.
+func (pss *PoolServiceStorage) TotalNVMe() uint64 {
```

Comment:
will be useful for system storage query

Reply:
Yep, that's what I was thinking. I figured I'd add this now because I was already in there, and having this in place will give us something to build on.